NebulaPatcher/Patches/Dynamic/GameHistoryData_Patch.cs

```diff
@@ -141,5 +141,26 @@ public static bool Prefix5()
             ///Wait for the authoritative packet for dequeing tech in multiplayer for clients
             return !SimulatedWorld.Initialized || LocalPlayer.IsMasterClient || GameDataHistoryManager.IsIncomingRequest;
         }
+
+        [HarmonyPrefix]
+        [HarmonyPatch("UnlockTech")]
+        public static bool Prefix6()
+        {
+            //Wait for the authoritative packet for unlocking tech features in multiplayer for clients
+            return !SimulatedWorld.Initialized || LocalPlayer.IsMasterClient || GameDataHistoryManager.IsIncomingRequest;
+        }
+
+        [HarmonyPrefix]
+        [HarmonyPatch("RemoveTechInQueue")]
+        public static void Prefix7(int index, out int __state)
+        {
```

Comment:
Is there a missed singleplayer chekc here?

Reply:
I've added if (**SimulatedWorld.Initialized &&** LocalPlayer.IsMasterClient), does this address your concern or do i need to add something else?